lib/spack/spack/cmd/env.py

```diff
@@ -724,6 +726,54 @@ def env_depfile(args):
         sys.stdout.write(makefile)
 
 
+def env_image_setup_parser(subparser):
+    """generate an OCI compatible container image the already installed environment"""
+    subparser.add_argument("--base-image")
+    subparser.add_argument("--force", default=False, action="store_true")
+    subparser.add_argument("--tag", "-t", required=True)
+    subparser.add_argument("mirror", type=arguments.mirror_name_or_url)
+
+
+def env_image(args):
+    image_ref = spack.oci.oci.image_from_mirror(args.mirror)
+    env = spack.cmd.require_active_env("env image")
+    roots = env.concrete_roots()
+    specs = env.all_specs()
+
+    base_image_ref: Optional[ImageReference] = (
+        ImageReference.from_string(args.base_image) if args.base_image else None
+    )
+
+    with tempfile.TemporaryDirectory(dir=spack.stage.get_stage_root()) as tmpdir:
+        with bc._make_pool() as pool:
+            skipped, base_images, checksums = bc._push_oci(args, image_ref, specs, tmpdir, pool)
+
+        architecture = bc._archspec_to_gooarch(roots[0])
```

Comment:
Here we are using functions from another command. Can we move these functions in some core package, since they are used in two places?

Reply:
yeah, let me move that somewhere.